extensions/amp-video-service/0.1/amp-video-service.js

```diff
@@ -76,14 +146,160 @@ export class VideoService {
 
   /**
    * @param {!AmpElement} unusedVideo
-   * @param {!../../../src/observable.Observable<boolean>} unusedObservable
+   * @param {!Observable<boolean>} unusedObservable
    */
   delegateAutoplay(unusedVideo, unusedObservable) {
     warnUnimplemented('Autoplay delegation');
   }
 }
 
 
+/**
+ * Handler for a registered video component.
+ * @visibleForTesting
+ */
+export class VideoEntry {
+
+  /**
+   * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+   * @param {!VideoService} videoService
+   * @param {!../../../src/video-interface.VideoInterface} video
+   */
+  constructor(ampdoc, videoService, video) {
+
+    /** @private @const {!../../../src/service/ampdoc-impl.AmpDoc} */
+    this.ampdoc_ = ampdoc;
+
+    /** @private @const {!VideoService} */
+    this.service_ = videoService;
+
+    /** @private @const {!../../../src/video-interface.VideoInterface} */
+    this.video_ = video;
+
+    /** @private {boolean} */
+    this.isPlaying_ = false;
+  }
+
+  /**
+   * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+   * @param {!VideoService} videoService
+   * @param {!../../../src/video-interface.VideoInterface} video
+   */
+  static create(ampdoc, videoService, video) {
+    const entry = new VideoEntry(ampdoc, videoService, video);
+    // Called here as opposed to constructor so that tests can stub VideoEntry
+    // methods before install.
+    entry.install();
+    return entry;
+  }
+
+  /** @param {function()} handler */
+  onPlaybackTick(handler) {
+    this.service_.onTick(() => {
+      if (!this.isPlaying_) {
+        return;
+      }
+      handler();
+    });
+  }
+
+  /** */
+  install() {
```

Comment:
nit: can become private now

Reply:
Hmm, I don't think so, since it's called from a static context.